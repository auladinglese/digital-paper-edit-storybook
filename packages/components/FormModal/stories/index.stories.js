import React from 'react';
import { storiesOf } from '@storybook/react';
import { actions } from '@storybook/addon-actions';
import StoryRouter from 'storybook-react-router';
import FormModal from '../index.js';

const modalItems = [ {
  id: 1,
  itemType: 'project',
  showModal: true,
  title: 'Example Project Title',
  description: 'This is a sample card description. This is fun!',
  url: '/projects/1/transcripts/1234',
  modalTitle: 'Edit Project',
}, {
  showModal: true,
  modalTitle: 'New Project',
  id: 2
}, {
  projectId: 123,
  title: '',
  description: '',
  uploadCompleted: true,
  showModal: true,
  modalTitle: 'New Transcript',
  id: 3,
  type: 'transcript'
} ];

const modalActions = actions({ handleSaveForm: 'Form saved' });

storiesOf('Form Modal', module)
  .addDecorator(StoryRouter())
  .add('Edit project', () => {
    return (
      <section style={ { height: '90vh', overflow: 'scroll' } }>
        <FormModal
          { ...modalActions }
          { ...modalItems[0] }
        />
      </section>
    );
  })
  .add('New project', () => {
    return (
      <section style={ { height: '90vh', overflow: 'scroll' } }>
        <FormModal
          { ...modalActions }
          { ...modalItems[1] }
        />
      </section>
    );
  })
  .add('New Transcript', () => {
    return (
      <section style={ { height: '90vh', overflow: 'scroll' } }>
        <FormModal
          { ...modalActions }
          { ...modalItems[2] }
        />
      </section>
    );
  });
