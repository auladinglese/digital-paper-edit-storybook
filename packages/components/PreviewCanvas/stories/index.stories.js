import React from 'react';

import { storiesOf } from '@storybook/react';
import PreviewCanvas from '../index.js';

const playlist = [

];

storiesOf('PreviewCanvas', module)
  .add('Default', () =>
    <PreviewCanvas
      playlist={ playlist }
    />
  );
