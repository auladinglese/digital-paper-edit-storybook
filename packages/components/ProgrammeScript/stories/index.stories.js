import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import ProgrammeScript from '../index.js';

export const handleUpdate = action('Handle reorder');

const programme = {
  elements:
  [
    {
      type: 'title',
      text: 'An immense Achievement'
    },
    {
      type: 'paper-cut',
      id: 1,
      speaker: 'Mr Loud',
      words: [ { text:'Greatest day of my life was when I wrote this text.', start: 0, end: 1 } ]
    },
    {
      type: 'note',
      text: 'Maybe a little bit obnoxious'
    },
    {
      type: 'paper-cut',
      id: 2,
      speaker: 'Mrs Loud',
      words: [ { text:'Greatest day of my life was when I spoke this text.', start: 0, end: 1 } ]
    },
    {
      type: 'voice-over',
      text: 'link: wonderful times of the Loud family'
    },
  ],
  title: 'Something'
};
const annotation = {
  'id':'1idcjw29xii80000ird74yb19swa',
  'start':14.38,
  'end':18.14,
  'labelId':'129cjw29xii80000ird74yb19swa',
  'note':'optional example text description for an annotation - TEST 2'
};

const transcripts = [
  {
    'id':'19cjw29xii80000ird74yb19swa',
    'description':'Transcript zero description',
    'url':'https://download.ted.com/talks/KateDarling_2018S-950k.mp4',
    'clipName':'KateDarling_2018S-950k.mp4',
    'status':'done',
    'transcript':{
      'words':[
        { 'id':0, 'start':13.02, 'end':13.17, 'text':'There' },
        { 'id':1, 'start':13.17, 'end':13.38, 'text':'is' },
        { 'id':2, 'start':13.38, 'end':13.44, 'text':'a' },
        { 'id':3, 'start':13.44, 'end':13.86, 'text':'day.' },
        { 'id':4, 'start':13.86, 'end':14.13, 'text':'About' },
        { 'id':5, 'start':14.13, 'end':14.38, 'text':'ten' },
        { 'id':6, 'start':14.38, 'end':14.61, 'text':'years', 'annotation':annotation },
        { 'id':7, 'start':14.61, 'end':15.15, 'text':'ago', 'annotation':annotation },
        { 'id':8, 'start':15.47, 'end':15.59, 'text':'when', 'annotation':annotation },
        { 'id':9, 'start':15.68, 'end':15.84, 'text':'I', 'annotation':annotation },
        { 'id':10, 'start':15.86, 'end':16.19, 'text':'asked', 'annotation':annotation },
        { 'id':11, 'start':16.19, 'end':16.28, 'text':'a', 'annotation':annotation },
        { 'id':12, 'start':16.28, 'end':16.65, 'text':'friend', 'annotation':annotation },
        { 'id':13, 'start':16.65, 'end':16.74, 'text':'to', 'annotation':annotation },
        { 'id':14, 'start':16.74, 'end':17.2, 'text':'hold', 'annotation':annotation },
        { 'id':15, 'start':17.23, 'end':17.33, 'text':'a', 'annotation':annotation },
        { 'id':16, 'start':17.33, 'end':17.63, 'text':'baby', 'annotation':annotation },
        { 'id':17, 'start':17.63, 'end':18.14, 'text':'dinosaur', 'annotation':annotation },
        { 'id':18, 'start':18.14, 'end':18.59, 'text':'robot' },
        { 'id':19, 'start':18.72, 'end':19.17, 'text':'upside' },
        { 'id':20, 'start':19.17, 'end':19.58, 'text':'down.' },
        { 'id':21, 'start':21.83, 'end':21.9, 'text':'It' },
        { 'id':22, 'start':21.95, 'end':22.09, 'text':'was' },
        { 'id':23, 'start':22.1, 'end':22.22, 'text':'a' },
        { 'id':24, 'start':22.23, 'end':22.68, 'text':'toy' },
        { 'id':25, 'start':22.72, 'end':22.99, 'text':'called' },
        { 'id':26, 'start':23.03, 'end':23.39, 'text':'plea.' },
      ],
      'paragraphs':[
        { 'id':0, 'start':13.02, 'end':13.86, 'speaker':'TBC 0' },
        { 'id':1, 'start':13.86, 'end':19.58, 'speaker':'TBC 1' },
        { 'id':2, 'start':21.83, 'end':23.39, 'speaker':'test' },
      ]
    },
    'projectTitle':'Project title One!',
    'transcriptTitle':'Kate Darling Ted Talk',
    'annotations':[ annotation ]
  }
];

storiesOf('ProgrammeScript', module)
  .add('Default', () =>
    <ProgrammeScript
      programme={ programme }
      transcripts={ transcripts }
      handleUpdate={ handleUpdate }
    />
  );
