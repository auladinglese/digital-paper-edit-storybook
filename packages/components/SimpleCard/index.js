import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Card from 'react-bootstrap/Card';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import { LinkContainer } from 'react-router-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faTrash,
  faPen
} from '@fortawesome/free-solid-svg-icons';
import 'bootstrap-css-only/css/bootstrap.css';

class SimpleCard extends Component {
    handleDelete = () => {
      const confirmDeleteText = "Click OK if you wish to delete or cancel if you don't";
      const cancelDeleteText = 'All is good, it was not deleted';
      const confirmationPrompt = confirm(confirmDeleteText);

      if (confirmationPrompt) {
        this.props.handleDelete ? this.props.handleDelete(this.props.id) : alert(cancelDeleteText);
      }
    };

    handleEdit = () => {
      this.props.handleEdit(this.props.id);
    }

    render() {

      return (
        <Card style={ { width: '100%', marginBottom: '1em' } }>
          <Card.Body>
            <Row>
              <LinkContainer to={ `${ this.props.url }` } style={ { cursor: 'pointer' } }>
                <Col xs={ 8 } sm={ 10 }>
                  <Card.Title>
                    {this.props.title}
                  </Card.Title>
                </Col>
              </LinkContainer>
              <Col xs={ 2 } sm={ 1 }>
                <Card.Link>
                  <Button
                    onClick={ this.handleEdit }
                    variant="outline-secondary"
                    size="sm"
                    aria-label="Edit button"
                  >
                    <FontAwesomeIcon icon={ faPen } />
                  </Button>
                </Card.Link>
              </Col>
              <Col xs={ 2 } sm={ 1 }>
                <Card.Link>
                  <Button
                    onClick={ this.handleDelete }
                    variant="outline-secondary"
                    size="sm"
                    aria-label="Delete button"
                  >
                    <FontAwesomeIcon icon={ faTrash } />
                  </Button>
                </Card.Link>
              </Col>
            </Row>
            <Row>
              <Col xs={ 10 } sm={ 11 }>
                <Card.Subtitle className="mb-2 text-muted">
                  {this.props.description}
                </Card.Subtitle>
              </Col>
            </Row>
          </Card.Body>
        </Card>
      );
    }
};

SimpleCard.propTypes = {
  key: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string,
  handleEdit: PropTypes.func,
  handleDelete: PropTypes.func,
  showLinkPath: PropTypes.func
};

SimpleCard.defaultProps = {
  key: 'key_1234',
  id: '1234',
  title: 'Default Title String',
  description: 'This is a default description string',
  handleEdit: () => {
    console.log('Edit button clicked');
  },
  handleDelete: () => {
    console.log('Delete button clicked');
  },
};

export default SimpleCard;
