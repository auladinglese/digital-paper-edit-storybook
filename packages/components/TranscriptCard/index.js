import React from 'react';
import PropTypes from 'prop-types';
import Card from 'react-bootstrap/Card';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import Alert from 'react-bootstrap/Alert';
import Spinner from 'react-bootstrap/Spinner';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faTrash,
  faCheck,
  faExclamationTriangle,
  faPen
} from '@fortawesome/free-solid-svg-icons';

const TranscriptCard = ({ transcriptItem, handleEdit, handleDelete }) => {

  const handleDeleteClick = () => {
    const confirmDeleteText = "Click OK if you wish to delete or cancel if you don't";
    const cancelDeleteText = 'All is good, it was not deleted';

    const confirmationPrompt = confirm(confirmDeleteText);

    if (confirmationPrompt) {
      return handleDelete ? handleDelete(transcriptItem.id) : alert(cancelDeleteText);
    }
  };

  const handleEditClick = () => {
    handleEdit(transcriptItem.id);
  };

  const setStatus = () => {
    if (transcriptItem.status === 'error') {
      return 'danger';
    };
    if (transcriptItem.status === 'in-progress') {
      return 'info';
    };
    if (transcriptItem.status === 'done') {
      return 'success';
    };
  };

  const status = setStatus();

  const setDescription = () => {
    switch (status) {
    case 'info':
      return <Badge variant="info">In progress</Badge>;
    case 'success':
      return <Badge variant="success">Success</Badge>;
    case 'danger':
      return (
        <>
          <Alert variant="danger">
            <FontAwesomeIcon icon={ faExclamationTriangle } />{' '}
            {transcriptItem.errorMessage}
          </Alert>
          <Badge variant="danger">Error</Badge>
        </>
      );
    }
  };

  const setStatusIcon = () => {
    switch (status) {
    case 'info':
      return (
        <Button variant="info" size="sm" disabled>
          <Spinner
            as="span"
            animation="border"
            size="sm"
            role="status"
            aria-hidden="true"
          />
        </Button>
      );
    case 'danger':
      return (
        <Button variant="danger" size="sm" disabled>
          <FontAwesomeIcon icon={ faExclamationTriangle } />
        </Button>
      );
    case 'success':
      return (
        <Button variant="success" size="sm" disabled>
          <FontAwesomeIcon icon={ faCheck } />
        </Button>
      );
    }
  };

  const description = setDescription();
  const statusIcon = setStatusIcon();
  const borderStatus = status && status === 'danger' ? 'danger' : null;
  const title = status && status === 'success' ? <a href={ `${ transcriptItem.url }` }> {transcriptItem.title}</a> : transcriptItem.title;

  return (
    <Card
      border={ borderStatus }
      style={ { width: '100%', marginBottom: '2em' } }
    >
      <Card.Body>
        <Row>
          <Col xs={ 12 } sm={ 9 }>
            <Card.Title>
              {transcriptItem.icon ? transcriptItem.icon : ''} {title}
            </Card.Title>
          </Col>
          <Col xs={ 2 } sm={ 1 }>
            <Card.Link>
              <Button
                onClick={ () => handleEditClick() }
                variant="outline-secondary"
                size="sm"
              >
                <FontAwesomeIcon icon={ faPen } />
              </Button>
            </Card.Link>
          </Col>
          <Col xs={ 2 } sm={ 1 }>
            <Card.Link>
              <Button
                onClick={ () => handleDeleteClick() }
                variant="outline-secondary"
                size="sm"
              >
                <FontAwesomeIcon icon={ faTrash } />
              </Button>
            </Card.Link>
          </Col>
          <Col xs={ 1 }>
            {statusIcon}
          </Col>
        </Row>
        <Row>
          <Col xs={ 12 }>
            <Card.Subtitle className="mb-2 text-muted">
              {transcriptItem.subtitle}
            </Card.Subtitle>
          </Col>
        </Row>
        <Row>
          <Col xs={ 12 }>
            {description}
          </Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

TranscriptCard.propTypes = {
  transcriptItem: PropTypes.object.isRequired,
  handleEdit: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired
};

TranscriptCard.defaultProps = {
  transcriptItem: {
    key: 'key_1234',
    id: '1234',
    title: 'Default Title String',
    description: 'This is a default description string',
    subtitle: 'This is a default subtitle',
    status: 'done',
    url: '/test/path/here',
  },
  handleEdit: () => {
    console.log('Edit button clicked');
  },
  handleDelete: () => {
    console.log('Delete button clicked');
  }
};
export default TranscriptCard;