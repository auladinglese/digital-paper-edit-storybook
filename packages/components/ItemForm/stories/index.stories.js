import React from 'react';

import { storiesOf } from '@storybook/react';
import StoryRouter from 'storybook-react-router';
import ItemForm from '../index.js';

import { modalItems, modalActions } from '../../FormModal/stories/index.stories.js';

storiesOf('Item Form', module)
  .addDecorator(StoryRouter())
  .add('Edit Project', () => {
    return (
      <section style={ { height: '90vh', overflow: 'scroll' } }>
        <ItemForm
          modalActions={ modalActions }
          modalItem={ modalItems[0] }
        />
      </section>
    );
  });