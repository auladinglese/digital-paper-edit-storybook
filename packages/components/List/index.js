import React, { useState } from 'react';
import addons, { mockChannel } from '@storybook/addons';
import PropTypes from 'prop-types';
import SimpleCard from '../SimpleCard';
import SearchBar from './SearchBar';
import TranscriptCard from '../TranscriptCard';

const List = (props) => {

  addons.setChannel(mockChannel());

  const [ items, setItems ] = useState(props.items);

  const includesText = (text, subsetText) => {
    return text.toLowerCase().includes(subsetText.toLowerCase().trim());
  };

  const handleDeleteItem = (itemId) => {
    const updatedList = items.filter((item) => {
      return item.id !== itemId;
    });
    props.handleDelete(itemId);
    setItems(updatedList);
  };

  const handleDisplay = (item, searchText) => {
    if (
      includesText(item.title, searchText) ||
            includesText(item.description, searchText)
    ) {
      item.display = true;
    } else {
      item.display = false;
    }

    return item;
  };

  const handleSearchItem = searchText => {
    const results = items.filter(item => handleDisplay(item, searchText));
    setItems(results);
  };

  const listItems = items.map((item) => {
    if (item.display && item.status) {
      return (
        <TranscriptCard
          { ...item }
          handleEdit={ props.handleEdit(item.id) }
          handleDelete={ handleDeleteItem }
        />
      );
    }
    else if (item.display) {
      return (
        <SimpleCard
          { ...item }
          handleEdit={ props.handleEdit(item.id) }
          handleDelete={ handleDeleteItem }
        />
      );}

    return null;
  }).filter(item => {
    return item !== null;
  });

  return (
    <section style={ { height: '75vh', overflow: 'scroll' } }>
      {items !== null && items.length !== 0 ? <SearchBar handleSearch={ handleSearchItem }/> : null}
      {listItems}
    </section>
  );
};

List.propTypes = {
  items: PropTypes.array.isRequired,
  handleEdit: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

List.defaultProps = {
  items: [
    {
      id: '1234',
      key: 'abc123',
      title: 'Sample Simple Card Title One',
      description: 'This is a sample card description. This is fun!',
      display: true,
      url: '/projects/1/transcripts/5678'
    }
  ],
  handleEdit: () => {
    console.log('Edit button clicked');
  },
  handleDelete: () => {
    console.log('Delete button clicked');
  },
};

export default List;
