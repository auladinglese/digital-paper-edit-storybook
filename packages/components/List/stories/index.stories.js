import React from 'react';

import { storiesOf } from '@storybook/react';
import StoryRouter from 'storybook-react-router';
import List from '../index.js';

import { item, cardActions } from '../../SimpleCard/stories/index.stories.js';
import { searchActions } from '../../SearchBar/stories/index.stories.js';

export const items = [{
    ...item,
    id: '1234',
    key: 'abc123',
    title: 'Sample Simple Card Title One',
    description: 'This is a sample card description. This is fun!',
    display: true,
}, {...item,
    id: '5678',
    key: 'def456',
    title: 'Sample Simple Card Title Two',
    description: 'This is a sample card description. This is fun!',
    display: true,
    url: 'projects/1/transcripts/'
}];

storiesOf('List', module)
    .addDecorator(StoryRouter())
    .add('default', () => 
            <List
                items={items}
                {...cardActions}
                {...searchActions}
            />
        );