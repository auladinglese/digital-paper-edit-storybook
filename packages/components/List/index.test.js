import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import List from './index.js';
import SimpleCard from '../SimpleCard';
import TranscriptCard from '../TranscriptCard';
import { items, transItems, searchActions } from './stories/index.stories';
import { cardActions } from '../SimpleCard/stories/index.stories';

describe('List component', () => {
  const onDelete = sinon.spy();
  const simpleCardList = shallow(<List items={ items } { ...cardActions } { ...searchActions } />);
  const transcriptCardList = shallow(<List { ...cardActions } { ...searchActions } items={ transItems } />);
  const mountedTranscriptCardList = mount(<MemoryRouter><List handleDelete={ onDelete } items={ transItems } /></MemoryRouter>);

  it('Should render correctly with Simple Cards and Transcript Cards', () => {
    expect(simpleCardList).toMatchSnapshot();
    expect(transcriptCardList).toMatchSnapshot();
  });

  it('Should correctly render multiple SimpleCard components', () => {
    expect(simpleCardList.find(SimpleCard).length).toBe(2);
  });

  it('Should correctly render multiple TranscriptCard components', () => {
    expect(transcriptCardList.find(TranscriptCard).length).toBe(3);
    expect(mountedTranscriptCardList.find(TranscriptCard).length).toBe(3);
    expect(mountedTranscriptCardList.find('Button').find({ variant: 'success' }).length).toBe(1);
    expect(mountedTranscriptCardList.find('Button').find({ variant: 'info' }).length).toBe(1);
    expect(mountedTranscriptCardList.find('Button').find({ variant: 'danger' }).length).toBe(1);
  });

  it('Should handle updating after a Simple Card is deleted', () => {
    window.confirm = jest.fn(() => true);
    const clickedComponent = mount(<MemoryRouter><List handleDelete={ onDelete } items={ items }/></MemoryRouter>);
    expect(clickedComponent.find(SimpleCard).length).toBe(2);
    clickedComponent.find('.delete-button').first().simulate('click');
    expect(clickedComponent.find(SimpleCard).length).toBe(1);
  });

  it('Should handle updating after a Transcript Card is deleted', () => {
    window.confirm = jest.fn(() => true);
    expect(mountedTranscriptCardList.find(TranscriptCard).length).toBe(3);
    mountedTranscriptCardList.find('.delete-button').first().simulate('click');
    mountedTranscriptCardList.find('.delete-button').first().simulate('click');
    expect(mountedTranscriptCardList.find(TranscriptCard).length).toBe(1);
  });

  describe('Search component', () => {
    //   Add tests for SearchBar functionality
  });
});