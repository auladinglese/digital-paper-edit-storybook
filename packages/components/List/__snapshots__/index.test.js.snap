// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Should render correctly with Simple Cards and Transcript Cards 1`] = `
<section
  style={
    Object {
      "height": "75vh",
      "overflow": "scroll",
    }
  }
>
  <SearchBar
    handleSearch={[Function]}
  />
  <SimpleCard
    description="This is a sample card description. This is fun!"
    display={true}
    handleDelete={[Function]}
    handleEdit={[Function]}
    id="1234"
    key="abc123"
    title="Sample Simple Card Title One"
    url="/projects/1/transcripts/1234"
  />
  <SimpleCard
    description="This is a sample card description. This is fun!"
    display={true}
    handleDelete={[Function]}
    handleEdit={[Function]}
    id="5678"
    key="def456"
    title="Sample Simple Card Title Two"
    url="/projects/1/transcripts/5678"
  />
</section>
`;

exports[`List component Should render correctly with Simple Cards and Transcript Cards 2`] = `
<section
  style={
    Object {
      "height": "75vh",
      "overflow": "scroll",
    }
  }
>
  <SearchBar
    handleSearch={[Function]}
  />
  <TranscriptCard
    description="This transcript has finished processing."
    display={true}
    handleDelete={[Function]}
    handleEdit={[Function]}
    id={1}
    key="transcript_key_1"
    status="done"
    subtitle="This transcript has finished processing."
    title="Title - Done Transcript"
    url="/projects/1/transcripts/"
  />
  <TranscriptCard
    description="This transcript is still being generated."
    display={true}
    handleDelete={[Function]}
    handleEdit={[Function]}
    id={2}
    key="transcript_key_2"
    status="in-progress"
    subtitle="This transcript is still being generated."
    title="Title - In Progress Transcript"
    url="/projects/1/transcripts/"
  />
  <TranscriptCard
    description="Transcript generation failed for this card."
    display={true}
    errorMessage="Something has gone wrong with your transcription"
    handleDelete={[Function]}
    handleEdit={[Function]}
    id={3}
    key="transcript_key_3"
    status="error"
    subtitle="Transcript generation failed for this card."
    title="Title - Error Transcript"
    url="/projects/1/transcripts/"
  />
</section>
`;
